src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -919,5 +919,80 @@ public static Task ReplayWarningsAndErrorsAsync(LockFile lockFile, ILogger logge
 
             return logger.LogMessagesAsync(logMessages);
         }
+
+        /// <summary>
+        /// Determines the restore style of a project.
+        /// </summary>
+        /// <param name="restoreProjectStyle">An optional user supplied restore style.</param>
+        /// <param name="hasPackageReferenceItems">A <see cref="bool"/> indicating whether or not the project has any PackageReference items.</param>
+        /// <param name="projectJsonPath">An optional path to the project's project.json file.</param>
+        /// <param name="projectDirectory">The full path to the project directory.</param>
+        /// <param name="projectName">The name of the project file.</param>
+        /// <param name="log">An <see cref="ILogger"/> object used to log messages.</param>
+        /// <returns>A <see cref="Tuple{ProjectStyle, bool}"/> containing the project style and a value indicating if the project is using a style that is compatible with PackageReference.</returns>
+        public static (ProjectStyle ProjectStyle, bool PackageReferenceCompatibleProjectStyle) GetProjectRestoreStyle(string restoreProjectStyle, bool hasPackageReferenceItems, string projectJsonPath, string projectDirectory, string projectName, ILogger log)
+        {
+            ProjectStyle projectStyle = ProjectStyle.Unknown;
+
+            // Allow a user to override by setting RestoreProjectStyle in the project.
+            if (!string.IsNullOrWhiteSpace(restoreProjectStyle))
+            {
+                if (!Enum.TryParse(restoreProjectStyle, out ProjectStyle userSuppliedProjectStyle))
```

Comment:
This seems like a breaking change.  As I understand it, MSBuild string comparisons are case-insensitive by default, but this overload of `Enum.TryParse(…)` is case sensitive.  That means, if a user explicitly sets their `RestoreProjectStyle` property like so:
```XML
<RestoreProjectStyle>packagereference</RestoreProjectStyle>
```
...then this parse would fail.  I think you need to use the other overload and pass `ignoreCase: true` to have parity.

Reply:
Great catch, I've updated the unit test and the parsing logic